filebeat/harvester/log.go

```diff
@@ -99,6 +99,13 @@ func (h *Harvester) Harvest() {
 			event.Text = &text
 			event.JSONFields = message.Fields
 			event.EventMetadata = h.config.EventMetadata
+			if message.Fields != nil {
```

Comment:
Not 100% sure I can follow. The current events look as following if fields is empty:

```
{
  "@timestamp": "2016-08-17T07:25:53.294Z",
  "beat": {
    "hostname": "ruflin",
    "name": "ruflin"
  },
  "fields": {},
  "input_type": "log",
  "message": "This exceeds the buffer",
  "offset": 24,
  "source": "/Users/ruflin/Dev/gopath/src/github.com/elastic/beats/filebeat/build/system-tests/run/test_harvester.Test.test_exceed_buffer/log/test.log",
  "type": "log"
}
```
I would expect the publisher / output to remove "fields" if it len(fields) == 0. If this is not the case yet, my question is what is the best place to implement this as I don't think this is filebeat specific.